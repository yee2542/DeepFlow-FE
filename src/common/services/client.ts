import { ApolloClient, createHttpLink, HttpLink, InMemoryCache } from '@apollo/client';

const endpoint = new HttpLink({
  uri: process.env.APP_GRAPHQL_ENDPOINT || 'http://localhost:5000/graphql',
});

export const client = new ApolloClient({
  uri: process.env.APP_GRAPHQL_ENDPOINT || 'http://localhost:5000/graphql',
  cache: new InMemoryCache(),
});
