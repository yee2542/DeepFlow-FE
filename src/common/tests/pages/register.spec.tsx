import FormCenterLayout from 'Modules/user/components/Layouts/FormCenter';
import { mount, shallow } from 'enzyme';
import React from 'react';
import RegisterPage from 'Modules/user/pages/register';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from 'Stores/index';
import { ApolloProvider } from '@apollo/client';
import { client } from 'Services/client';
import { MockedProvider } from '@apollo/client/testing';

const mocks = [];
describe('Pages/Register test', () => {
  it('Render correctly', () => {
    const wrap = renderer
      .create(
        <Provider store={store}>
          <MockedProvider mocks={mocks}>
            <RegisterPage />
          </MockedProvider>
        </Provider>,
      )
      .toJSON();
    expect(wrap).toMatchSnapshot();
  });

  it('RegisterPage should be exisiting', () => {
    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage />
        </MockedProvider>
      </Provider>,
    );
    expect(wrap.exists(RegisterPage)).toBe(true);
  });

  it('Should contain FormCenter Layout', () => {
    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage />
        </MockedProvider>
      </Provider>,
    );
    expect(wrap.exists(FormCenterLayout)).toBe(true);
  });

  it('Should have name, email & password field / register btn', () => {
    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage />
        </MockedProvider>
      </Provider>,
    );
    const nameField = wrap.find('#name');
    const emailField = wrap.find('#email');
    const passwordField = wrap.find('#password');
    const registerButton = wrap.find('#register-btn');

    expect(nameField.exists()).toBe(true);
    expect(emailField.exists()).toBe(true);
    expect(passwordField.exists()).toBe(true);
    expect(registerButton.exists()).toBe(true);
  });

  it('Should call onError when fill only email', () => {
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(1));
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(0));

    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage onSubmit={onSubmit} onError={onError} />
        </MockedProvider>
      </Provider>,
    );
    const emailField = wrap.find('input[id="email"]');
    emailField.simulate('change', { target: { value: 'Hello@email.com' } });

    wrap.find('#register-btn').at(0).simulate('submit');
  });

  it('Should call onError when not fulfill', () => {
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(1));
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(0));

    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage onSubmit={onSubmit} onError={onError} />
        </MockedProvider>
      </Provider>,
    );
    const nameField = wrap.find('input[id="name"]');
    const emailField = wrap.find('input[id="email"]');

    nameField.simulate('change', { target: { value: 'John Doe' } });
    emailField.simulate('change', { target: { value: 'Hello@email.com' } });

    wrap.find('#register-btn').at(0).simulate('submit');
  });

  it('Should call onError when not a email', () => {
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(1));
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(0));

    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage onSubmit={onSubmit} onError={onError} />
        </MockedProvider>
      </Provider>,
    );
    const emailField = wrap.find('input[id="email"]');
    emailField.simulate('change', { target: { value: 'Hello' } });

    wrap.find('#register-btn').at(0).simulate('submit');
  });

  it('Should call onSubmit when fullfill', () => {
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(1));
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(0));

    const wrap = mount(
      <Provider store={store}>
        <MockedProvider mocks={mocks}>
          <RegisterPage onSubmit={onSubmit} onError={onError} />
        </MockedProvider>
      </Provider>,
    );
    const emailField = wrap.find('input[id="email"]');
    const passwordField = wrap.find('input[id="password"]');
    const nameField = wrap.find('input[id="name"]');

    emailField.simulate('change', { target: { value: 'Hello' } });
    passwordField.simulate('change', { target: { value: '1234' } });
    nameField.simulate('change', { target: { value: 'John Doe' } });

    wrap.find('#register-btn').at(0).simulate('submit');
  });
});
