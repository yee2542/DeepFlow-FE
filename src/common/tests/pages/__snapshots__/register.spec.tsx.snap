// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Register test Render correctly 1`] = `
<div
  className="styled__Container-sc-1un2s9o-0 YXrJt"
>
  <div
    className="ant-row ant-row-center"
    style={
      Object {
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
      }
    }
  >
    <div
      className="ant-col ant-col-xs-0 ant-col-sm-0 ant-col-md-0 ant-col-lg-12"
      style={Object {}}
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "166.66666666666669%",
            }
          }
        />
        <img
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="ant-col ant-col-xs-20 ant-col-sm-20 ant-col-md-20 ant-col-lg-12"
      style={Object {}}
    >
      <div
        className="styled__FormContainer-sc-1un2s9o-1 eFwkdR"
      >
        <div
          className="ant-row ant-row-center ant-row-middle"
          style={
            Object {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
            }
          }
        >
          <div
            className="ant-col ant-col-8"
            style={Object {}}
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjEyMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style={
                    Object {
                      "display": "block",
                      "maxWidth": "100%",
                    }
                  }
                />
              </div>
              <img
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-row-center"
          style={
            Object {
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
            }
          }
        >
          <div
            className="ant-col ant-col-16 ant-col-xl-14"
            style={Object {}}
          >
            <form
              className="ant-form ant-form-vertical ant-form-large"
              id="register-form"
              onSubmit={[Function]}
            >
              <div
                className="ant-row ant-form-item"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <div
                  className="ant-col ant-col-24 ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <span
                        className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                        onMouseUp={[Function]}
                      >
                        <span
                          className="ant-input-prefix"
                        >
                          <span
                            aria-label="user"
                            className="anticon anticon-user"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="user"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          className="ant-input ant-input-lg"
                          id="name"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="name"
                          style={null}
                          type="text"
                          value=""
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-form-item"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <div
                  className="ant-col ant-col-24 ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <span
                        className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                        onMouseUp={[Function]}
                      >
                        <span
                          className="ant-input-prefix"
                        >
                          <span
                            aria-label="mail"
                            className="anticon anticon-mail"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="mail"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          className="ant-input ant-input-lg"
                          id="email"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="email"
                          style={null}
                          type="text"
                          value=""
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-form-item"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <div
                  className="ant-col ant-col-24 ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <span
                        className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                        onMouseUp={[Function]}
                      >
                        <span
                          className="ant-input-prefix"
                        >
                          <span
                            aria-label="lock"
                            className="anticon anticon-lock"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="lock"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          className="ant-input ant-input-lg"
                          id="password"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="Password"
                          style={null}
                          type="password"
                          value=""
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-form-item"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <div
                  className="ant-col ant-col-24 ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <button
                        className="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
                        id="register-btn"
                        onClick={[Function]}
                        type="submit"
                      >
                        <span>
                          Register
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-row-center"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <div
                  className="ant-col ant-col-10"
                  style={Object {}}
                >
                  <hr />
                </div>
                <div
                  className="ant-col ant-col-4"
                  style={Object {}}
                >
                  <p
                    className="global__P-m421n3-0 iUPqqF"
                  >
                    or
                  </p>
                </div>
                <div
                  className="ant-col ant-col-10"
                  style={Object {}}
                >
                  <hr />
                </div>
              </div>
              <div
                className="ant-row ant-form-item"
                style={
                  Object {
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              >
                <div
                  className="ant-col ant-col-24 ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <p
                        className="global__P-m421n3-0 iUPqqF"
                      >
                        Already have an account?
                        <a
                          href="/login"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                           Log in
                        </a>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
